tests/testthat/test-uncount.R

```diff
@@ -22,7 +22,33 @@ test_that("expands constants and expressions", {
   expect_equal(uncount(df, 1 + 1), df[c(1, 1), ])
 })
 
-test_that("must evaluate to numeric", {
+
+test_that("works with groups", {
+  df <- tibble(g = 1, x = 1, w = 1) %>% dplyr::group_by(g)
+  expect_equal(uncount(df, w), df %>% dplyr::select(-w))
+})
+
+test_that("doesn't remove grouping variable", {
+  df <- tibble(g = 1, x = 1, w = 1) %>% dplyr::group_by(g)
+
+  expect_message(uncount(df, g)) %>%
+    expect_equal(df)
+})
```

Comment:
I added it to the documentation of `.remove`. Not sure whether you'd prefer it there or in the details section.

I am not totally sure whether this is the correct behaviour but to me it feels more consistent with `dplyr::select()`. The feeling of surprise originally came from something like in the following example:

``` r
library(tidyr)

df <- tibble(g = c("a", "a", "b"), x = 1, w = c(1, 2, 1)) %>% 
  dplyr::group_by(g, w)

uncount(df, w)
#> # A tibble: 4 x 2
#> # Groups:   g [2]
#>   g         x
#>   <chr> <dbl>
#> 1 a         1
#> 2 a         1
#> 3 a         1
#> 4 b         1
uncount(df, w, .remove = FALSE)
#> # A tibble: 4 x 3
#> # Groups:   g, w [3]
#>   g         x     w
#>   <chr> <dbl> <dbl>
#> 1 a         1     1
#> 2 a         1     2
#> 3 a         1     2
#> 4 b         1     1
uncount(df, w + 0)
#> # A tibble: 4 x 3
#> # Groups:   g, w [3]
#>   g         x     w
#>   <chr> <dbl> <dbl>
#> 1 a         1     1
#> 2 a         1     2
#> 3 a         1     2
#> 4 b         1     1
```

<sup>Created on 2020-12-08 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

@hadley what's your opinion on that?

Reply:
I think uncounting by a grouping variable is going to be relatively uncommon, and there's no clear best answer (since the two most applicable rules clash), so I think that argues for keeping the behaviour as is.